[Flaky Test] ShuffleKerberizedHdfsStorageUtilsTest failed when hostname is not "localhost"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)


### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.


### Describe the flaky test

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 7.856 s <<< FAILURE! - in org.apache.uniffle.storage.util.ShuffleKerberizedHdfsStorageUtilsTest
[ERROR] testUploadFile{File}  Time elapsed: 0.044 s  <<< ERROR!
java.lang.IllegalArgumentException: Wrong FS: hdfs://localhost:40981/test, expected: hdfs://devcloud:40981
	at org.apache.hadoop.fs.FileSystem.checkPath(FileSystem.java:665)
	at org.apache.hadoop.hdfs.DistributedFileSystem.getPathName(DistributedFileSystem.java:214)
	at org.apache.hadoop.hdfs.DistributedFileSystem$27.doCall(DistributedFileSystem.java:1440)
	at org.apache.hadoop.hdfs.DistributedFileSystem$27.doCall(DistributedFileSystem.java:1437)
	at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
	at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1437)
	at org.apache.hadoop.fs.FileSystem.isFile(FileSystem.java:1463)
	at org.apache.uniffle.storage.handler.impl.HdfsFileWriter.initStream(HdfsFileWriter.java:59)
	at org.apache.uniffle.storage.handler.impl.HdfsFileWriter.<init>(HdfsFileWriter.java:54)
	at org.apache.uniffle.storage.util.ShuffleHdfsStorageUtilsTest.createAndRunCases(ShuffleHdfsStorageUtilsTest.java:57)
	at org.apache.uniffle.storage.util.ShuffleKerberizedHdfsStorageUtilsTest.testUploadFile(ShuffleKerberizedHdfsStorageUtilsTest.java:39)
```

### Actions URL

Run locally

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

--- a/common/src/test/java/org/apache/uniffle/common/KerberizedHdfs.java
+++ b/common/src/test/java/org/apache/uniffle/common/KerberizedHdfs.java
@@ -259,7 +259,7 @@ public class KerberizedHdfs implements Serializable {
   }
 
   public String getSchemeAndAuthorityPrefix() {
-    return "hdfs://" + kerberizedDfsCluster.getNameNode().getNameNodeAddressHostPortString() + "/";
+    return kerberizedDfsCluster.getURI().toString();
   }
 
   public Configuration getConf() throws IOException {

--- a/common/src/test/java/org/apache/uniffle/common/KerberizedHdfs.java
+++ b/common/src/test/java/org/apache/uniffle/common/KerberizedHdfs.java
@@ -259,7 +259,7 @@ public class KerberizedHdfs implements Serializable {
   }
 
   public String getSchemeAndAuthorityPrefix() {
-    return kerberizedDfsCluster.getURI().toString();
+    return kerberizedDfsCluster.getURI().toString() + "/";
   }
 
   public Configuration getConf() throws IOException {

--- a/common/src/test/java/org/apache/uniffle/common/KerberizedHdfs.java
+++ b/common/src/test/java/org/apache/uniffle/common/KerberizedHdfs.java
@@ -259,7 +259,7 @@ public class KerberizedHdfs implements Serializable {
   }
 
   public String getSchemeAndAuthorityPrefix() {
-    return String.format("hdfs://localhost:%s/", kerberizedDfsCluster.getNameNodePort());
+    return "hdfs://" + kerberizedDfsCluster.getNameNode().getNameNodeAddressHostPortString() + "/";
   }
 
   public Configuration getConf() throws IOException {

--- a/server/src/test/java/org/apache/uniffle/server/ShuffleFlushManagerTest.java
+++ b/server/src/test/java/org/apache/uniffle/server/ShuffleFlushManagerTest.java
@@ -153,7 +153,7 @@ public class ShuffleFlushManagerTest extends HdfsTestBase {
     StorageManager storageManager =
         StorageManagerFactory.getInstance().createStorageManager(shuffleServerConf);
     storageManager.registerRemoteStorage(appId, remoteStorage);
-    String storageHost = "localhost";
+    String storageHost = cluster.getURI().getHost();
     assertEquals(0.0, ShuffleServerMetrics.counterRemoteStorageTotalWrite.get(storageHost).get(), 0.5);
     assertEquals(0.0, ShuffleServerMetrics.counterRemoteStorageRetryWrite.get(storageHost).get(), 0.5);
     assertEquals(0.0, ShuffleServerMetrics.counterRemoteStorageFailedWrite.get(storageHost).get(), 0.5);

--- a/storage/src/test/java/org/apache/uniffle/storage/HdfsTestBase.java
+++ b/storage/src/test/java/org/apache/uniffle/storage/HdfsTestBase.java
@@ -47,7 +47,7 @@ public class HdfsTestBase implements Serializable {
     conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR,
         baseDir.getAbsolutePath());
     cluster = (new MiniDFSCluster.Builder(conf)).build();
-    HDFS_URI = "hdfs://localhost:" + cluster.getNameNodePort() + "/";
+    HDFS_URI = "hdfs://" + cluster.getNameNode().getNameNodeAddressHostPortString() + "/";
     fs = (new Path(HDFS_URI)).getFileSystem(conf);
   }
 

--- a/storage/src/test/java/org/apache/uniffle/storage/HdfsTestBase.java
+++ b/storage/src/test/java/org/apache/uniffle/storage/HdfsTestBase.java
@@ -47,7 +47,7 @@ public class HdfsTestBase implements Serializable {
     conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR,
         baseDir.getAbsolutePath());
     cluster = (new MiniDFSCluster.Builder(conf)).build();
-    HDFS_URI = cluster.getURI().toString();
+    HDFS_URI = cluster.getURI().toString() + "/";
     fs = (new Path(HDFS_URI)).getFileSystem(conf);
   }
 

--- a/storage/src/test/java/org/apache/uniffle/storage/HdfsTestBase.java
+++ b/storage/src/test/java/org/apache/uniffle/storage/HdfsTestBase.java
@@ -47,7 +47,7 @@ public class HdfsTestBase implements Serializable {
     conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR,
         baseDir.getAbsolutePath());
     cluster = (new MiniDFSCluster.Builder(conf)).build();
-    HDFS_URI = "hdfs://" + cluster.getNameNode().getNameNodeAddressHostPortString() + "/";
+    HDFS_URI = cluster.getURI().toString();
     fs = (new Path(HDFS_URI)).getFileSystem(conf);
   }
 

--- a/storage/src/test/java/org/apache/uniffle/storage/util/ShuffleHdfsStorageUtilsTest.java
+++ b/storage/src/test/java/org/apache/uniffle/storage/util/ShuffleHdfsStorageUtilsTest.java
@@ -53,7 +53,7 @@ public class ShuffleHdfsStorageUtilsTest extends HdfsTestBase {
     dataOut.write(buf);
     dataOut.close();
     fileOut.close();
-    String path = clusterPathPrefix + "/test";
+    String path = clusterPathPrefix + "test";
     HdfsFileWriter writer = new HdfsFileWriter(fileSystem, new Path(path), hadoopConf);
     long size = ShuffleStorageUtils.uploadFile(file, writer, 1024);
     assertEquals(2096, size);

--- a/storage/src/test/java/org/apache/uniffle/storage/util/ShuffleHdfsStorageUtilsTest.java
+++ b/storage/src/test/java/org/apache/uniffle/storage/util/ShuffleHdfsStorageUtilsTest.java
@@ -32,7 +32,6 @@ import org.apache.uniffle.storage.HdfsTestBase;
 import org.apache.uniffle.storage.handler.impl.HdfsFileWriter;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.fail;
 
 public class ShuffleHdfsStorageUtilsTest extends HdfsTestBase {
 
@@ -46,27 +45,20 @@ public class ShuffleHdfsStorageUtilsTest extends HdfsTestBase {
       FileSystem fileSystem,
       String clusterPathPrefix,
       Configuration hadoopConf) throws Exception {
-    FileOutputStream fileOut = null;
-    DataOutputStream dataOut = null;
-    try {
-      File file = new File(tempDir, "test");
-      fileOut = new FileOutputStream(file);
-      dataOut = new DataOutputStream(fileOut);
-      byte[] buf = new byte[2096];
-      new Random().nextBytes(buf);
-      dataOut.write(buf);
-      dataOut.close();
-      fileOut.close();
-      String path = clusterPathPrefix + "test";
-      HdfsFileWriter writer = new HdfsFileWriter(fileSystem, new Path(path), hadoopConf);
-      long size = ShuffleStorageUtils.uploadFile(file, writer, 1024);
-      assertEquals(2096, size);
-      size = ShuffleStorageUtils.uploadFile(file, writer, 100);
-      assertEquals(2096, size);
-      writer.close();
-    } catch (Exception e) {
-      e.printStackTrace();
-      fail();
-    }
+    File file = new File(tempDir, "test");
+    FileOutputStream fileOut = new FileOutputStream(file);
+    DataOutputStream dataOut = new DataOutputStream(fileOut);
+    byte[] buf = new byte[2096];
+    new Random().nextBytes(buf);
+    dataOut.write(buf);
+    dataOut.close();
+    fileOut.close();
+    String path = clusterPathPrefix + "test";
+    HdfsFileWriter writer = new HdfsFileWriter(fileSystem, new Path(path), hadoopConf);
+    long size = ShuffleStorageUtils.uploadFile(file, writer, 1024);
+    assertEquals(2096, size);
+    size = ShuffleStorageUtils.uploadFile(file, writer, 100);
+    assertEquals(2096, size);
+    writer.close();
   }
 }

--- a/storage/src/test/java/org/apache/uniffle/storage/util/ShuffleHdfsStorageUtilsTest.java
+++ b/storage/src/test/java/org/apache/uniffle/storage/util/ShuffleHdfsStorageUtilsTest.java
@@ -53,7 +53,7 @@ public class ShuffleHdfsStorageUtilsTest extends HdfsTestBase {
     dataOut.write(buf);
     dataOut.close();
     fileOut.close();
-    String path = clusterPathPrefix + "test";
+    String path = clusterPathPrefix + "/test";
     HdfsFileWriter writer = new HdfsFileWriter(fileSystem, new Path(path), hadoopConf);
     long size = ShuffleStorageUtils.uploadFile(file, writer, 1024);
     assertEquals(2096, size);

